src/tools/rbd_mirror/image_replayer/PrepareLocalImageRequest.cc

```diff
@@ -124,9 +126,25 @@ void PrepareLocalImageRequest<I>::handle_get_mirror_info(int r) {
     return;
   }
 
-  // TODO save current mirror state to determine if we should
-  // delete a partially formed image
-  // (e.g. MIRROR_IMAGE_STATE_CREATING/DELETING)
+  if (m_mirror_image.state == cls::rbd::MIRROR_IMAGE_STATE_CREATING) {
+    dout(5) << "local image is still in creating state, issuing a removal"
+            << dendl;
+    auto ctx = new LambdaContext([this](int) {
+        finish(-ENOENT);
+    });
+    ImageDeleter<I>::trash_move(m_io_ctx, m_global_image_id,
+                                false, m_threads->work_queue, ctx);
```

Comment:
nit: I would prefer adding `move_to_trash` and `handle_move_to_trash` methods. And documenting the state in the state diagram.

Reply:
I did it, but not sure if the formatting of the comment in the state diagram is correct.